app/views.py

```diff
@@ -1268,34 +1273,27 @@ def folder_update(request, folder_id):
         return Response(settings, status.HTTP_200_OK)
     if request.method == "PUT":
         form = request.data
-        id = request.data["item_id"]
         if not form:
             return Response("Folder Data is Required", status.HTTP_400_BAD_REQUEST)
+        items = form["items"]
         old_folder = get_folder_object(folder_id)
         old_folder["folder_name"] = form["folder_name"]
-        old_folder["data"].append(
-            {"item_id:": form["item_id"], "item_type": form["item_type"]}
-        )
+        old_folder["data"].extend(items)
         updt_folder = json.loads(update_folder(folder_id, old_folder))
-        if request.data["item_type"] == "document":
```

Comment:
All these nested if statements not required, this logic should be takes outside this view function.
A view should only accept response and return the necessary response.. Keep business logic out of your views?

Reply:
The business logics here are small, and I didn't much find it necessary to create much external imports them, those i found excess were kept from it.